sdk/core/src/headers/mod.rs

```diff
@@ -178,13 +193,14 @@ impl HeaderName {
 
 impl From<&'static str> for HeaderName {
     fn from(s: &'static str) -> Self {
+        assert_eq!(s.to_lowercase(), s, "header names must be lower-case");
```

Comment:
This forces a canonicalization of header names to prevent issues such as what I found in data tables, where even after adding headers to the `prepare_request`, I found that data_tables was using `Content-Type` rather than `content-type`, which did show up in `Headers.get("content-type")`.  

By forcing a canonicalized form here, we prevent footguns later.